components/net/http_loader.rs

```diff
@@ -682,7 +673,8 @@ pub fn http_redirect_fetch(request: &mut Request,
     request.redirect_count += 1;
 
     // Step 7
-    let same_origin = location_url.origin()== request.current_url().origin();
+    // FIXME: Correctly use request's origin
+    let same_origin = location_url.origin() == request.current_url().origin();
```

Comment:
That shouldn't be neccessary either. All code paths that come from `RequestInit` should already have an `ImmutableOrigin` set for them.